### PR TITLE
Only make spock fetches if the data is needed to display the active page

### PR DESCRIPTION
**Pull request by jparklev**

### CODE REVIEW

File: src/pages/Polling.js [hunk 1]
```diff
@@ -340,11 +340,13 @@ class Polling extends React.Component {
       this.state.activeAccount !== prevProps.activeAccount ||
       (this.props.poll && prevProps.poll === undefined)
     ) {
-      await this.updateVotedPollOption();
+      this.props.pollDataInit(this.props.poll);
```
Comment (jparklev) [inline src/pages/Polling.js]: initiate poll data fetch from the polling details page based on the `pollId` of that page

File: src/reducers/polling.js [hunk 2]
```diff
@@ -312,7 +315,6 @@ export const pollsInit = () => async dispatch => {
               .service('smartContract')
               .getContract('POLLING').address;
             dispatch(addPoll(pollData));
-            dispatch(pollDataInit(pollData));
```
Comment (jparklev) [inline src/reducers/polling.js]: previously, this is where we auto-fetched poll data for each poll returned from `getAllWhiteListedPolls`

**Comment by tyler17** [comment]
When I run this locally, the polls don't load for at https://localhost:3000/polling

**Comment by jparklev** [comment]
@tyler17 if you look at the dev tools network tab on `/polling`, do you see any requests being made to the spock endpoint?

**Comment by tyler17** [comment]
@jparklev Ah I see a request made to fetch all polls but it's to `http://localhost:3001/v1`, instead of `https://gov-db.makerfoundation.com/api/v1`

**Comment by tyler17** [comment]
Ok don't mind me, I was linked to a local version of dai-plugin-governance that forced the UI to connect to local spock.  Its working now.

**Comment by tyler17** [comment]
I do notice that when I go to a specific poll page, voteOptionMkrWeights is called 3 times with the same parameters, and then a 4th time with a different block number

**Comment by jparklev** [comment]
> Maybe the sdk should cache the value returned from voteOptionMkrWeights

right, i had a similar thought. maybe we could memoize `getMkrSupport` on `pollId` & `blockNumber`

**Comment by tyler17** [comment]
Ok so this commit should do that: https://github.com/makerdao/dai-plugin-governance/compare/memoizeMkrSupport?expand=1
But as is it doesn't help because the frontend initiates all the getMkrSupport calls at the same time, so I think we'd have to update it to wait for 1 before calling the rest.

**Pull request merged**